### ISSUE TITLE
Google Drive Connection Errors in Linux

### ISSUE THREAD

**Issue by cmbr** [duplicati/duplicati#2097]
I have installed Duplicati - 2.0.1.31_canary_2016-11-03 on two computers. One is running Linux Mint 17.3 Cinnamon 64-bit. The other is running Linux Mint 18 Cinnamon 64-bit. I start Duplicati from the menu on both and the http://127.0.0.1:8200/ngax/index.html page opens in my default Firefox browser.

Then I start to Add a Backup. At step 2, Backup Destination, I select Proprietary > Google Drive as the storage type and enter the path on the server. I open the AuthID link and the Duplicati OAuth Handler modal pops up. I click the Google Drive Login button. I choose the GMail account associated with my Google Drive. I Allow Duplicati to have offline access. The new AuthID is successfully entered into the Add Backup form.

Now I test the connection. After some time I receive this error message:
_Failed to connect: The server certificate had the error RemoteCertificateChainErrors and the hash <removed> ..._

Essentially the same error message (with different hash codes) appears on both computers.

On both computers, I open a terminal and issue this command:
mozroots --import --sync

On the Linux Mint 17.3 machine, new root certificates were added to the trust store. On Linux Mint 18, I receive this message:
Couldn't retrieve the file using the supplied information. It doesn't matter anyway. All attempts to test the connection to my Google Drive fail.

Earlier issue reports similar to this linked to the http://code.google.com/p/duplicati/wiki/LinuxHowto#Issues_with_,_Skydrive,_and_S3 page. That link is now redirected to the Duplicati github home page.

Is there a way to use Duplicati on Linux with Google Drive?

**Comment by kakoskin1** [comment]
Command mozroot has been deprecated. Currently mono uses system certificate store. Command to synchronise system certificates to mono is cert-sync. See for example: http://www.mono-project.com/docs/about-mono/releases/3.12.0/#cert-sync
Despite what above link says, I have never (on Fedora) had certificates synchronised automatically at to of installing mono. 

Advice to run mozroot should be replaced by advice to use cert-sync.

**Comment by kenkendk** [comment]
@kakoskin1 : Thanks for the link!

**Comment by barghest** [comment]
Hi,
I've run the command 
`cert-sync /etc/ssl/certs/ca-certificates.crt`
but still get the error message on Arch Linux

```
Linux Cert Store Sync - version 4.6.1.0
Synchronize local certs with certs from local Linux trust store.
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

I already trust 160, your new list has 160
Import process completed.
```

**Comment by Nerdy314159265** [comment]
I'm running on FreeNAS 9.10 and trying to install Duplicati in a jail. Even after running cert-sync, I also have the same issue.

**Comment by kenkendk** [comment]
I have updated the SSL wiki page:
https://github.com/duplicati/duplicati/wiki/SSL-TLS-support-in-Mono

For Linux Mint, make sure you have the repo from Xamarin, as the Debian Mono is quite old.

**Comment by kenkendk** [comment]
Closing for lack of activity

**Comment by JohnyBeGood** [comment]
Hi,

**Note:I could not reply to original post on these forums.**

I had below error on a new CentOS 7 install.
Tried all suggestions found online regarding installing and reinstalling "_ca-certificates-mono_" but I already had it installed and nothing else worked until I came across this how-to https://github.com/duplicati/duplicati/wiki/SSL-TLS-support-in-Mono
and after issuing this command "_sudo cert-sync /etc/pki/tls/certs/ca-bundle.crt_"  I was able to continue with my first backup. Hope it helps someone!

> Failed to connect: The server certificate had the error RemoteCertificateChainErrors
